Source/WebCore/page/scrolling/AsyncScrollingCoordinator.cpp

```diff
@@ -889,60 +887,58 @@ void AsyncScrollingCoordinator::setNodeLayers(ScrollingNodeID nodeID, const Node
 
     node->setLayer(nodeLayers.layer);
 
-    if (is<ScrollingStateScrollingNode>(node)) {
-        auto& scrollingNode = downcast<ScrollingStateScrollingNode>(*node);
-        scrollingNode.setScrollContainerLayer(nodeLayers.scrollContainerLayer);
-        scrollingNode.setScrolledContentsLayer(nodeLayers.scrolledContentsLayer);
-        scrollingNode.setHorizontalScrollbarLayer(nodeLayers.horizontalScrollbarLayer);
-        scrollingNode.setVerticalScrollbarLayer(nodeLayers.verticalScrollbarLayer);
-
-        if (is<ScrollingStateFrameScrollingNode>(node)) {
-            auto& frameScrollingNode = downcast<ScrollingStateFrameScrollingNode>(*node);
-            frameScrollingNode.setInsetClipLayer(nodeLayers.insetClipLayer);
-            frameScrollingNode.setCounterScrollingLayer(nodeLayers.counterScrollingLayer);
-            frameScrollingNode.setRootContentsLayer(nodeLayers.rootContentsLayer);
+    if (auto* scrollingNode = dynamicDowncast<ScrollingStateScrollingNode>(*node)) {
+        scrollingNode->setScrollContainerLayer(nodeLayers.scrollContainerLayer);
+        scrollingNode->setScrolledContentsLayer(nodeLayers.scrolledContentsLayer);
+        scrollingNode->setHorizontalScrollbarLayer(nodeLayers.horizontalScrollbarLayer);
+        scrollingNode->setVerticalScrollbarLayer(nodeLayers.verticalScrollbarLayer);
+
+        if (auto* frameScrollingNode = dynamicDowncast<ScrollingStateFrameScrollingNode>(*node)) {
```

Comment:
No reason not to pass `*scrollingNode` here. There is no perf benefit with the current `is<>()` implementation for this type but it's always good to use as tight typing as possible in general.